vengine/vengine/resource_management/loaders/ColliderLoader.cpp

```diff
@@ -0,0 +1,203 @@
+	{
+		glm::vec3 whd(
+			abs(mesh->mVertices[0].x) * 100,
+			abs(mesh->mVertices[0].z) * 100, 
+			abs(mesh->mVertices[0].y) * 100
+		);
+
+		return Collider::createBox(whd);
+	}
+	//else if (type == shapeType::Cone)//not supported by engine
+	//{
+	//}
+	//else if (type == shapeType::Cylinder)//not supported by engine
+	//{
+	//}
+	else if (type == ColType::CAPSULE)
+	{
+		float radius = aiVector3D(mesh->mVertices[0].x, 0, mesh->mVertices[0].z).Length();
+		float height = aiVector3D(0,mesh->mVertices[0].y, 0).Length() * 2;//vet ej om det ska vara 2 eller inte
+		return Collider::createCapsule(radius, height);
+	}
+	return Collider::createBox(glm::vec3(1));
+}
+
+std::vector<ColliderDataRes> ColliderLoader::loadCollisionShape(const std::string& modelFile)
+{
+	//this->setComponent<Collider>(camEntity, Collider::createBox(glm::vec3(1)));
+	const aiScene* scene = this->importer.ReadFile((modelFile).c_str(),
+		aiProcess_JoinIdenticalVertices 
+		| aiProcess_RemoveComponent 
+		| aiProcess_DropNormals  
+	);
+	
+	if (scene == nullptr)
+	{
+		Log::warning("Failed to load Collision (" + modelFile + ")");
+		return std::vector<ColliderDataRes>();
+	}
+
+	std::vector<ColliderDataRes> collisionList;
+
+	auto* node = scene->mRootNode;
+	auto scenes_meshes = std::span<aiMesh*>(scene->mMeshes, scene->mNumMeshes);
+	auto node_children = std::span<aiNode*>(node->mChildren, node->mNumChildren);
+	auto node_meshes = std::span<unsigned int>(node->mMeshes, node->mNumMeshes);
+
+	std::stack<aiNode*> nodeStack;
+	nodeStack.push(node);
+
+	while (!nodeStack.empty())
+	{
+		node_meshes = std::span<unsigned int>(node->mMeshes, node->mNumMeshes);
+		for (size_t j = 0; j < node->mNumMeshes; j++)
+		{
+			std::string meshName(scenes_meshes[node_meshes[j]]->mName.C_Str());
+			if (meshName.length() > 7 && meshName.substr(0, 8) == "Collider")
+			{
+				//get what kind of shape it is
+				ColType theShapeType = getShapeType(scenes_meshes[node_meshes[j]], meshName);
+				if (theShapeType == ColType::COLERROR)
+				{
+					Log::error("can't load the collision box");
+					return collisionList;
+				}
+				//get position and rotation of mesh
+				aiMatrix4x4 modelMatrix = node->mTransformation;
+				glm::mat4 transformation(
+					node->mTransformation.a1, node->mTransformation.b1, node->mTransformation.c1, node->mTransformation.d1, 
+					node->mTransformation.a2, node->mTransformation.b2, node->mTransformation.c2, node->mTransformation.d2,
+				    node->mTransformation.a3, node->mTransformation.b3, node->mTransformation.c3, node->mTransformation.d3,
+				    node->mTransformation.a4, node->mTransformation.b4, node->mTransformation.c4, node->mTransformation.d4
+				);
+				glm::vec3 scale;
+				glm::quat rotation;
+				glm::vec3 translation;
+				glm::vec3 skew;
+				glm::vec4 perspective;
+				glm::decompose(transformation, scale, rotation, translation, skew, perspective);
+				glm::vec3 eRotation = glm::eulerAngles(rotation) * (180.0f / 3.141592653589793238463f);
```

Comment:
glm has support to convert to radians and to degrees (glm::radians(), glm::degrees()) Use those or atleast save (180.0f / 3.141592653589793238463f) to variable instead of calculating ever loop iteration